src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -1507,8 +1505,13 @@ private static string GetBlobBaseUri(string blobUri)
                 replacedName = replacedName.Replace(".symbols", "", StringComparison.OrdinalIgnoreCase);
                 replacedName = replacedName.Replace(".nupkg", "", StringComparison.OrdinalIgnoreCase);
 
-                // finally, remove the version from the name
-                replacedName = replacedName.Replace($".{asset.Version}", "", StringComparison.OrdinalIgnoreCase);
+                // finally, remove the version from the name, which should be the last element
+                string versionSegmentToRemove = $".{asset.Version}";
+                if (!replacedName.EndsWith(versionSegmentToRemove))
+                {
+                    Logger.LogWarning($"Warning: Expected that '{asset.Name}', with package and path parts removed, would end in '{asset.Version}'. Instead was '{replacedName}'.");
+                }
+                replacedName = replacedName.Remove(replacedName.Length - versionSegmentToRemove.Length);
```

Comment:
This just removes the last n characters?

Reply:
Yeah, the version prefixed with '.'.

Since we're trying to compute the name of the package from the file, and the symbol package has the same file name format as a normal package, It should be ending with ".version". It's possible there is some other case which is why I put in the warning.